core/controllers/base_test.py

```diff
@@ -2257,29 +2295,15 @@ class MockUploadHandler(base.BaseHandler):
             }
         }
 
-        def post(self, entity_type: str, entity_id: str) -> None:
+        # Here we use MyPy ignore because the signature of 'get' method does not
```

Comment:
Shouldn't this be "post" method?

Also I'm wondering, why are all these mypy-ignores coming in now (why didn't they arise in earlier rounds of typing)? What changed and what's the reasoning behind this fix?

Reply:
> Shouldn't this be "post" method?

Yes, and this thing is fixed all over the file!

> Also I'm wondering, why are all these mypy-ignores coming in now

This issue is mentioned earlier in the doc, but after covering half the folder I realized that this issue is only occurring inside base_test.py, not in other files, and this is because here get method ( or other HTTP methods ) are directly compared with its superclass while in main files we have decorators that changes the function signature.

And I think it is guaranteed that in the future this error will not occur in any main file because every handler's HTTP method is going to have a decorator, and we have test for that:
https://github.com/oppia/oppia/blob/5471941da48071723b5f7e8684716584c08c6a1d/core/controllers/base_test.py#L1073

Reply:
Got it, thanks! That additional context is helpful and I think you should include it in the comment -- something like "... This happens because all handler methods in the main codebase have decorators which modify the function signature accordingly, but these methods in base_test.py do not."

Reply:
Comment added.